From 248ee85948f0cfa0c0d37cb424c5decf0e079288 Mon Sep 17 00:00:00 2001
From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Thu, 10 Jan 2013 12:06:20 +0300
Subject: [PATCH 318/858] [SCSI] bfa: fix strncpy() limiter in bfad_start_ops()

The closing parenthesis is in the wrong place so it takes the sizeof a
pointer instead of the sizeof the buffer minus one.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Acked-by: Anil Gurumurthy <agurumur@brocade.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 30ac379a6921b14f4c44e13875d39b8f8f478958)

Signed-off-by: Clark Williams <williams@redhat.com>
Signed-off-by: John Kacur <jkacur@redhat.com>
---
 drivers/scsi/bfa/bfad.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/bfa/bfad.c b/drivers/scsi/bfa/bfad.c
index e6bf126..a5f7690 100644
--- a/drivers/scsi/bfa/bfad.c
+++ b/drivers/scsi/bfa/bfad.c
@@ -1034,7 +1034,7 @@ bfad_start_ops(struct bfad_s *bfad) {
    sizeof(driver_info.host_os_patch) - 1);
 
  strncpy(driver_info.os_device_name, bfad->pci_name,
-  sizeof(driver_info.os_device_name - 1));
+  sizeof(driver_info.os_device_name) - 1);
 
  /* FCS driver info init */
  spin_lock_irqsave(&bfad->bfad_lock, flags);
-- 
1.8.3.1